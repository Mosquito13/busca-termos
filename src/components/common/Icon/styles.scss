.icon {
  height: 100%;
  width: 100%;

  &--extra-small {
    font-size: 1.5rem;
    height: 1.5rem;
  }

  &--small {
    font-size: 2rem;
    height: 2rem;
  }

  &--medium {
    font-size: 3rem;
    height: 3rem;
  }

  &--large {
    font-size: 6rem;
    height: 6rem;
  }

  svg {
    width: 100%;
    height: 100%;
  }
}
