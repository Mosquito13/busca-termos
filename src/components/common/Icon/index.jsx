import classNames from 'classnames';
import PropTypes from 'prop-types';

import './styles.scss';

const EXTRA_SMALL = 1;
const SMALL = 2;
const MEDIUM = 3;
const LARGE = 4;
const RESPONSIVE = 5;

const Icon = ({ size, children }) => {
  const classes = classNames(
    'icon',
    size === EXTRA_SMALL && 'icon--extra-small',
    size === SMALL && 'icon--small',
    size === MEDIUM && 'icon--medium',
    size === LARGE && 'icon--large'
  );

  return <div className={classes}>{children}</div>;
};

Icon.propTypes = {
  /**
   * Tamanho do botao
   */
  size: PropTypes.oneOf([null, EXTRA_SMALL, SMALL, MEDIUM, LARGE, RESPONSIVE])
};

Icon.defaultProps = {
  size: MEDIUM
};

export { EXTRA_SMALL, SMALL, MEDIUM, LARGE, RESPONSIVE };
export default Icon;
