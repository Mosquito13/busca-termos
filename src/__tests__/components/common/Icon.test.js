import renderer from 'react-test-renderer';
import { FiAlertOctagon } from 'react-icons/fi';

import Icon, {
  EXTRA_SMALL,
  SMALL,
  MEDIUM,
  LARGE,
  RESPONSIVE
} from '../../../components/common/Icon';

describe(`${Icon.name}`, () => {
  it('should render extra small', () => {
    const tree = renderer
      .create(
        <Icon size={EXTRA_SMALL}>
          <FiAlertOctagon />
        </Icon>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render small', () => {
    const tree = renderer
      .create(
        <Icon size={SMALL}>
          <FiAlertOctagon />
        </Icon>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render medium', () => {
    const tree = renderer
      .create(
        <Icon size={MEDIUM}>
          <FiAlertOctagon />
        </Icon>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render large', () => {
    const tree = renderer
      .create(
        <Icon size={LARGE}>
          <FiAlertOctagon />
        </Icon>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render responsive', () => {
    const tree = renderer
      .create(
        <Icon size={RESPONSIVE}>
          <FiAlertOctagon />
        </Icon>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
