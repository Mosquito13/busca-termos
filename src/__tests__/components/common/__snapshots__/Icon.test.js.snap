// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render extra small 1`] = `
<div
  className="icon icon--extra-small"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
</div>
`;

exports[`Icon should render large 1`] = `
<div
  className="icon icon--large"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
</div>
`;

exports[`Icon should render medium 1`] = `
<div
  className="icon icon--medium"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
</div>
`;

exports[`Icon should render responsive 1`] = `
<div
  className="icon"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
</div>
`;

exports[`Icon should render small 1`] = `
<div
  className="icon icon--small"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
</div>
`;
